src/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# Source files
+file(GLOB_RECURSE SOURCES_SC2LADDERSERVER "*.cpp" "*.h")
+
+# Include directories
+include_directories(SYSTEM
+    ${PROJECT_SOURCE_DIR}/s2client-api/include
+    ${PROJECT_SOURCE_DIR}/s2client-api/contrib/protobuf/src
+    ${PROJECT_BINARY_DIR}/s2client-api/generated
+    ${PROJECT_SOURCE_DIR}/rapidjson
+	if (WIN32) # Curl include for windows
```

Comment:
Please move this condition outside there is no benefit from embedding it into include_directories.

Reply:
That's a mistake by me: I meant for it to be like this:
```
# Include directories
include_directories(SYSTEM
    ${PROJECT_SOURCE_DIR}/s2client-api/include
    ${PROJECT_SOURCE_DIR}/s2client-api/contrib/protobuf/src
    ${PROJECT_BINARY_DIR}/s2client-api/generated
    ${PROJECT_SOURCE_DIR}/rapidjson
if (WIN32) # Curl include for windows
    ${PROJECT_SOURCE_DIR}/curl/include/curl
endif ()
)
```
Is this okay?

Reply:
It is a matter of taste really.

However regarding curl I would suggest to move this inclusion outside and group it together with the 'linking' part. It will simplify addition of cmake variable (the one which will enable/disable curl) in the future.

Reply:
Yeh, I think having a specific option to disable curl would be the best option.  I want to modify the curl install so it will work on all platforms too, so when we have that working, it should be available as an argument on all platforms

Reply:
Yes, agreed there.
I want to clean up and group what I can, but I'm prioritizing getting it all functional first before doing so. Otherwise I know I'll probably inadvertently break something.